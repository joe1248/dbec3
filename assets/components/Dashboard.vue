
<template>
    <div id="menu" style="position: relative; margin-top: 100px;">
        <p>
            <router-link :to="{name: 'ListOfConnections'}">List of connections</router-link>
        </p>
        <p>
            <router-link :to="{name: 'NewConnection'}">Add new connection</router-link>
        </p>
    </div>
</template>

<script lang="ts">
    // script after template helps coverage get to 100% instead of 0...
    export default {
        data() {
            return {};
        }
    }
</script>
