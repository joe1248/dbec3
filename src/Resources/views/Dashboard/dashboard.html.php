<!-- src/Resources/views/Security/login.html.php -->
<?php $title = 'DBEC Dashboard' ?>

<?php ob_start() ?>
    <?php if ($user): ?>
    <h1>Hi <?php echo $user->getUsername() ?></h1> <a href="/logout">Logout</a>
    <?php endif ?>

    <h1>Dashboard</h1>
    
    <h3>Connections</h3>
    <p>
        <a href="connections">List</a>
        <a href="connections/add">Add new</a>
    </p>
    
    <h3>Entities</h3>
    <p>
        <a href="entities">List</a>
        <a href="entities/add">Add new</a>
    </p>
    
<?php $content = ob_get_clean() ?>

<?php include '../templates/layout.php' ?>
