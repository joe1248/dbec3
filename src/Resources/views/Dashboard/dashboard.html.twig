{% extends 'base.html.twig' %}

{% block title %} Dashboard {% endblock %}

{% block stylesheets %}
    {{ parent() }}
    <link rel="stylesheet" href="{{ asset('build/app.css') }}">
{% endblock %}

{% block body %}
    <!-- TOP MENU = 1 setting icon and onmousehover : link to account settings, soon FK guesser and log out link  + Version number + occasional UPdate DBEC button.. --->

    <div class="nobr dbec_about_menu"><br><br>
        <!-- TOOLS BUTTON -->
        <ul class="jquery_menu_main" id="menu_settings" style="position:relative;display:inline-block;">
            <li>
                <img style="height:20px;width:21px;" src="/images/settings.jpg">
                <ul style="display:none;">
                    <li><a href="#" class="nobr" id="button_account_details">Account Settings</a></li>
                    <li><a href="#" id="button_new_logout" class="nobr">Log out</a></li>
                </ul>
            </li>
        </ul>

        <span id="auto_test_user_home_title">DBEC Version 1.0</span>
        <!-- DOCUMENTATION is about old veriosn of dbec, so maybe update it, or at least copy the intro on live web site-->
        <!--a href="#"  id="button_documentation" style="position:relative;display:inline-block;">
            <img style="height:24px;border:0;" src="../html_cdn/images_icon/help.jpg">
        </a-->
    </div>
    <div class="title_color nobr dbec_update_menu_button">
        <input type="button" class="small-button hidden" id="button_to_get_latest_dbec_version"
               value="Update to DBec xxx" />
        <span id="dbec_is_up_to_date"  class="color_inverted_1 hidden">Up to date ;)</span>
    </div>
    <div id="app">
        <Dashboard></Dashboard>
        <router-view></router-view>
    </div>


    <!--div id="home_tabs" class="ui-layout-center">
        <div id="home_tab_welcome" class="tab_content">
            <div class="col-group">
                <div>
                    <div><br><br>
                        <fieldset class="fieldset one_line hidden" style="position:relative;top:7px;height:60px;">
                            <input type="button" value="Add A DB Server" id="welcome_big_button_add_db_server" style="top:10px;" class="welcome_button_add_db_server">
                        </fieldset>
                        <fieldset class="fieldset one_line">
                            <legend class="legend">Basics</legend>
                            
                            <input type="button" value="Manage your servers" id="welcome_button_db_servers_list">
                            <input type="button" value="+" class="welcome_button_add_db_server" title="Add a DB server"> 
                            <br>
                            <br>
                            <input type="button" value="Create A New Entity" id="welcome_big_button_create_entity" class="welcome_button_create_entity">
                            
                            <div>
                                <input type="button" value="Manage your entities" id="welcome_button_entities_list">
                                <input type="button" value="+" class="welcome_button_create_entity" title="Create a new entity">
                            </div>
                            <br>
                            <input type="button" value="Database Explorer" id="welcome_button_database_explorer">
                        </fieldset>
                    </div>
                </div>
                
                <div><br><br>
                    <fieldset class="fieldset one_line">
                        <legend class="legend">Clone Data</legend>
                        <input type="button" value="one entire table" id="welcome_big_button_clone_entire_table" class="welcome_button_clone_entire_table">
                        <br>
                        <span id="div_list_of_entities_to_clone">...</span>
                    </fieldset>
                </div>
            
                <div style="clear:left;">
                </div>

                <div><br><br>
                    <fieldset class="fieldset one_line">
                        <legend class="legend">Recent Transfers</legend>
                        <span id="div_list_of_recent_transfers">...</span>
                    </fieldset>
                </div>
            </div>
        </div>
        <ul class="top_tabs">
            <li><a id="link_home_tab" href="#home_tab_welcome">Home</a></li>
        </ul>
    </div>
    <input type="hidden" id="new_registered_id" value="0">

-->
{% endblock %}

{% block javascripts %}
    {{ parent() }}
    <script type="text/javascript"  src="{{ asset('build/app.js') }}"></script>
{% endblock %}
