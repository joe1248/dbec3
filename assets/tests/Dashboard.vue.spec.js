import { mount, shallow } from '@vue/test-utils' // see https://vue-test-utils.vuejs.org
import Dashboard from './../components/Dashboard'
import {createRenderer} from "vue-server-renderer";

describe('DashboardComponent', () => {
    const wrapper = mount(Dashboard);

    test('is a Vue instance', () => {
        expect(wrapper.isVueInstance()).toBeTruthy()
    });

    test('should have no default data', function () {
        expect(wrapper.vm._data).toEqual({});
    });

    test('should have default html', function () {
        const wrapper = shallow(Dashboard);
        const renderer = createRenderer();
        renderer.renderToString(wrapper.vm, (err, str) => {
            if (err) throw new Error(err.toString());
            expect(str).toMatchSnapshot()
        })
    });

    test('should have no default props', function () {
        expect(wrapper.props()).toEqual({});
    });
});