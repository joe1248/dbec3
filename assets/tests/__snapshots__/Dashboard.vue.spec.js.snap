// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardComponent should have default html 1`] = `
<div id="menu" style="position:relative;margin-top:100px;">
    <p>
        <router-link to="[object Object]">List of connections</router-link>
    </p>
    <p>
        <router-link to="[object Object]">Add new connection</router-link>
    </p>
</div>
`;
